synapseclient/models/wiki.py

```diff
@@ -1 +1,1018 @@
+            wiki_id=self.id,
+            wiki_version=self.wiki_version,
+            synapse_client=synapse_client,
+        )
+
+    @otel_trace_method(
+        method_to_trace_name=lambda self, **kwargs: f"Get_Attachment_URL: Owner ID {self.owner_id}, Wiki ID {self.id}, File Name {kwargs['file_name']}"
+    )
+    async def get_attachment_async(
+        self,
+        file_name: str,
+        *,
+        download_file: bool = True,
+        download_location: Optional[str] = None,
+        redirect: Optional[bool] = False,
+        synapse_client: Optional["Synapse"] = None,
+    ) -> Union[str, None]:
+        """
+        Download the wiki page attachment to a local file or return the URL.
+
+        Arguments:
+            file_name: The name of the file to get.
+            download_file: Whether associated files should be downloaded. Default is True.
+            download_location: The directory to download the file to. Required if download_file is True.
+            redirect: When set to false, the URL will be returned as text/plain instead of redirecting. Default is False.
+            synapse_client: Optionally provide a Synapse client.
+        Returns:
+            If download_file is True, the attachment file will be downloaded to the download_location. Otherwise, the URL will be returned.
+        Raises:
+            ValueError: If owner_id or id is not provided.
+        """
+        if not self.owner_id:
+            raise ValueError("Must provide owner_id to get attachment URL.")
+        if not self.id:
+            raise ValueError("Must provide id to get attachment URL.")
+        if not file_name:
+            raise ValueError("Must provide file_name to get attachment URL.")
+
+        client = Synapse.get_client(synapse_client=synapse_client)
+        attachment_url = await get_attachment_url(
+            owner_id=self.owner_id,
+            wiki_id=self.id,
+            file_name=file_name,
+            wiki_version=self.wiki_version,
+            redirect=redirect,
+            synapse_client=client,
+        )
+
+        if download_file:
+            if not download_location:
+                raise ValueError("Must provide download_location to download a file.")
+
+            # construct PresignedUrlInfo for downloading
+            presigned_url_info = PresignedUrlInfo(
+                url=attachment_url,
+                file_name=file_name,
+                expiration_utc=_pre_signed_url_expiration_time(attachment_url),
+            )
+            filehandle_dict = await get_attachment_handles(
+                owner_id=self.owner_id,
+                wiki_id=self.id,
+                wiki_version=self.wiki_version,
+                synapse_client=client,
+            )
+            # check the file_size
+            file_size = int(WikiPage._get_file_size(filehandle_dict, file_name))
+            # use single thread download if file size < 8 MiB
+            if file_size < 8388608:
```

Comment:
I think there is a constant you can import instead of using a specific number on this line.